shared/chat/conversation/input-area/normal/container.tsx

```diff
@@ -61,50 +61,42 @@ let _channelSuggestions: Array<{channelname: string; teamname?: string}> = noCha
 const getChannelSuggestions = (
   state: Container.TypedState,
   teamname: string,
-  teamID: TeamsTypes.TeamID,
+  _: TeamsTypes.TeamID,
   convID?: Types.ConversationIDKey
 ) => {
   if (!teamname) {
     // this is an impteam, so get mutual teams from state
     if (!convID) {
       return noChannel
     }
-    const mutualTeams = state.chat2.mutualTeamMap.get(convID) ?? []
-    return mutualTeams.reduce<Array<{channelname: string; teamname: string}>>((arr, id) => {
-      const teamname = TeamsConstants.getTeamNameFromID(state, id)
-      if (!teamname) {
+    const mutualTeams = (state.chat2.mutualTeamMap.get(convID) ?? []).map(teamID =>
+      TeamsConstants.getTeamNameFromID(state, teamID)
+    )
+    // TODO: maybe we shouldn't rely on this inboxlayout being around?
+    return (state.chat2.inboxLayout?.bigTeams ?? []).reduce<Array<{channelname: string; teamname: string}>>(
+      (arr, t) => {
+        if (t.state === RPCChatTypes.UIInboxBigTeamRowTyp.channel) {
+          if (mutualTeams.includes(t.channel.teamname)) {
+            arr.push({channelname: t.channel.channelname, teamname: t.channel.teamname})
+          }
+        }
         return arr
-      }
-      const channels: TeamsTypes.ChannelInfo[] = Array.from(
-        state.teams.teamIDToChannelInfos.get(id)?.values() ?? []
-      )
-
-      return arr.concat(
-        [...channels.values()].map(conv => ({
-          channelname: conv.channelname,
-          teamname,
-        }))
-      )
-    }, [])
+      },
+      []
+    )
   }
-  // First try channelinfos (all channels in a team), then try inbox (the
-  // partial list of channels that you have joined).
-  const convs = state.teams.teamIDToChannelInfos.get(teamID)
-  let suggestions: Array<{channelname: string}>
-  if (convs) {
-    suggestions = [...convs.values()].map(conv => ({
-      channelname: conv.channelname,
-    }))
-  } else {
-    suggestions = (state.chat2.inboxLayout?.bigTeams ?? []).reduce<Array<{channelname: string}>>((arr, t) => {
+  // TODO: get all the channels in the team, too, for this
```

Comment:
Why not use the hook here for this?

Reply:
You can only call hooks from within components or other hooks, and this is called from a container.

Reply:
Ah mhmm

Reply:
From what I can tell this is status quo except that channel suggestions in a team only show the channels you're in. We should ticket / fix if possible.